library/socket/tcpserver/sysaccept_spec.rb

```diff
@@ -1,2 +1,36 @@
 require File.expand_path('../../../../spec_helper', __FILE__)
 require File.expand_path('../../fixtures/classes', __FILE__)
+
+require 'socket'
+
+describe "TCPServer#sysaccept" do
+  before :each do
+    @server = TCPServer.new(SocketSpecs.hostname, SocketSpecs.port)
+  end
+
+  after :each do
+    @server.close unless @server.closed?
+  end
+
+  it 'blocks if no connections' do
+    begin
+      thread = Thread.new { @server.sysaccept }
+      thread.join(0.1).should == nil
+    ensure
+      Thread.kill(thread)
```

Comment:
Did not know there was the class method `kill` :)
You should probably join the thread after killing it:
```ruby
thread.kill
thread.join
```
Or use the new matcher :)